templates/README.md

```diff
@@ -0,0 +1,141 @@
+# Concordium Smart Contract Templates
+
+This repo helps you to get a new Concordium smart contract project up and running quickly by leveraging the pre-existing templates in this git repository. There are several smart contract templates available in this repository (currently a `default` template and a `cis2-nft` template). For generating the smart contracts from the templates, you need the `cargo-concordium` crate. `Cargo-concordium` can be installed as described in the following guide:
+
+[Cargo-concordium setup](https://developer.concordium.software/en/mainnet/smart-contracts/guides/setup-tools.html#setup-tools)
+
+You also need the `cargo-generate` crate. `Cargo-generate` can be installed by running the following command:
+
+```
+cargo install --locked cargo-generate
+```
+
+Here is an example of using the templates:
+
+## Examples
+
+### Generating a new project
+
+<img src="./gif_recordings/init.gif" width="700" height="450"/>
```

Comment:
Could we move all of the images from the top-level into a directory like `resources`?